onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -955,10 +955,30 @@ def aten_choose_qparams_optimized(
     raise NotImplementedError()
 
 
-def aten_chunk(self: TensorType, chunks: int, dim: int = 0) -> TensorType:
+@torch_op("aten::chunk")
+def aten_chunk(self: TTensor, chunks: INT64, dim: int = 0) -> TTensor:
     """chunk(Tensor(a -> *) self, int chunks, int dim=0) -> Tensor(a)[]"""
 
-    raise NotImplementedError()
+    neg_1 = op.Constant(value_ints=[-1])
+    # get size of specified dim
+    self_shape = op.Shape(self)
+    dim_size = op.Gather(self_shape, dim, axis=0)
+    # cal size/chunk, get number of data in one chunk
+    num_per_chunk = op.Div(dim_size, chunks)
+    if op.Mod(dim_size, chunks) > 0:  # type: ignore[operator]
+        # cannot be divisible, then num_in_chunk + 1
+        num_per_chunk = op.Add(num_per_chunk, 1)
```

Comment:
I wonder if using Where will be more efficient or not

Reply:
Even better: `op.Cast(op.Mod(dim_size, chunks) > 0, to=INT64.value) + num_per_chunk`

Reply:
The take away for me was `If` tends to be slow and should be replaced with other things when possible.